### PR TITLE
Feat/57 create product service interface

### PR DESCRIPTION
**Pull request by mkucmus**
I've created basic product service, which contains three methods:

```
  getProductsIds: () => Promise<SearchResult<string[]>>;
  getProducts: () => Promise<SearchResult<Product[]>>;
  getProduct: (productId: string) => Promise<SearchResult<Partial<Product>>>;
```

The according tests were created as well. 

I proposed a new files structure to keep services tidy - add some new folders.

### CODE REVIEW

File: packages/shopware-6-client/src/endpoints.ts [hunk 1]
```diff
@@ -1 +1,6 @@
 export const CATEGORY_ENDPOINT = "/category";
+
+/** Shopware API path for products  */
+export const PRODUCT_ENDPOINT = "/product";
+/** Base path for search-ids queries */
+export const SEARCH_IDS_ENDPOINT = "/search-ids";
```
[the remark this answers is not in the document]
Comment (patzick) [reply packages/shopware-6-client/src/endpoints.ts]: @mkucmus please move this as strings into methods for now and @rmakara will change this with endpoints PR :)

File: packages/shopware-6-client/src/interfaces/response/SingleResult.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface SingleResult<T> {
```
Comment (patzick) [inline packages/shopware-6-client/src/interfaces/response/SingleResult.ts]: I think we don't need this interface

File: packages/shopware-6-client/src/services/productService.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+export interface ProductService {
+  getProductsIds: () => Promise<SearchResult<string[]>>;
+  getProducts: () => Promise<SearchResult<Product[]>>;
+  getProduct: (productId: string) => Promise<SearchResult<Partial<Product>>>;
+}
+
+/**
+ * Get default amount of products
+ *
+ * @returns Promise<SearchResult<Product[]>>
```
Comment (patzick) [inline packages/shopware-6-client/src/services/productService.ts]:
```suggestion
```

File: packages/shopware-6-client/src/services/productService.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  const resp = await axios.post(
+    `${config.endpoint}${SEARCH_IDS_ENDPOINT}${PRODUCT_ENDPOINT}`
+  );
+  return resp.data;
+};
+
+/**
+ * Get default amount of products' ids
+ *
+ * @returns Promise<SearchResult<string[]>>
```
Comment (patzick) [inline packages/shopware-6-client/src/services/productService.ts]:
```suggestion
```

File: packages/shopware-6-client/src/services/productService.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ */
+const getProducts = async function(): Promise<SearchResult<Product[]>> {
+  const resp = await axios.get(`${config.endpoint}${PRODUCT_ENDPOINT}`);
+  return resp.data;
+};
+
+/**
+ * Get the product with passed productId
+ *
+ * @param productId
```
Comment (patzick) [inline packages/shopware-6-client/src/services/productService.ts]:
```suggestion
```

File: packages/shopware-6-client/src/services/productService.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+ * @returns Promise<SearchResult<Partial<Product>>>
```
Comment (patzick) [inline packages/shopware-6-client/src/services/productService.ts]:
```suggestion
```

File: packages/shopware-6-client/src/services/productService.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+ */
+const getProduct = async function(
+  productId: string
+): Promise<SearchResult<Partial<Product>>> {
```
Comment (patzick) [inline packages/shopware-6-client/src/services/productService.ts]:
```suggestion
): Promise<Product> {
```

File: packages/shopware-6-client/src/endpoints.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkucmus) [reply packages/shopware-6-client/src/endpoints.ts]: If it's not a problem I've already chosen the first option :)

**Pull request merged**